### PR TITLE
Feature: sprint checkin form submission status for user

### PR DESCRIPTION
**Pull request by Ajen07**
# Description

added check in submission status in the response, the sprintId will be in sprintCheckin list if the user has submitted a check in form for the sprint

## Issue link
[
Fixes # (issue)](https://app.clickup.com/t/86b00h4gx)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- unit tests
unit-test-passed

- e2e tests
e2e-test-passed

- swagger
swaager

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Review by cherylli — COMMENTED** [review]
edit: move to review comment

**Review by cherylli — CHANGES_REQUESTED** [review]
we shouldn't need to add any field to the table, we can retrieve the data without added another field to the table

you can retrieve the data from sprint check in table and append it to the /me endpoint

we don't want to add extra/unnecessary columns to tables, we need to make use of the relations

I guess one thing to keep in mind is we don't want to create duplicate data just for the purpose of retrieval.

====
So to reiterate,

in the fetch user detail endpoint `users/me`,

fetch the users voyageTeamMemberId, use this id to query sprintCheckin table, append this data to the response, so something like this

```ts
prisma.FormResponseCheckin.findMany({
where: voyageTeamMemberId {
id: { in: voyageTeamMemberId}
}, seleect:{
  sprintId: true
}
```

where voyageTeamMemberId is the list of Id from the user table


image

**Review by JoshuaHinman — APPROVED** [review]
Tests passed and checkins displayed in response - nice job!  It would have been helpful for the users/me route to be identified in the Description or Tested section - at first I assumed it was the sprints/checkin route

**Review by cherylli — CHANGES_REQUESTED** [review]

**Comment by cherylli** [comment]
sorry I forgot to submit my reviews which I did a couple of days ago, so they were pending

File: prisma/migrations/20240507131015_add_sprint_checkin_for_user/migration.sql
[the remark this answers is not in the document]
Comment (Ajen07) [reply prisma/migrations/20240507131015_add_sprint_checkin_for_user/migration.sql:1]: Actually all my up migrations are in a single commit but the down migration and other changes are in a single commit.So is there any way I can remove both the migration file ?

**Comment by Ajen07** [comment]
> Tests passed and checkins displayed in response - nice job! It would have been helpful for the users/me route to be identified in the Description or Tested section - at first I assumed it was the sprints/checkin route

Got it ! .From the next time , I will make my descriptions more elaborative about the issue and its solutions .

File: prisma/migrations/20240507131015_add_sprint_checkin_for_user/migration.sql
[the remark this answers is not in the document]
Comment (cherylli) [reply prisma/migrations/20240507131015_add_sprint_checkin_for_user/migration.sql:1]: I think you can just delete both migration folders locally and commit then they won't show in the changes

[the remark this answers is not in the document]
Comment (Ajen07) [reply prisma/migrations/20240507131015_add_sprint_checkin_for_user/migration.sql:1]: Ok I will delete them locally and push the changes.

**Review by cherylli — APPROVED** [review]

**Review by timDeHof — APPROVED** [review]
Great job! All tests passes and swagger endpoint responses with a 201.

**Pull request merged**